src/procedures/proc_pagerank.c

```diff
@@ -5,41 +5,61 @@
 */
 
 #include "proc_pagerank.h"
+#include "../RG.h"
 #include "../value.h"
 #include "../util/arr.h"
 #include "../query_ctx.h"
 #include "../util/rmalloc.h"
 #include "../graph/graphcontext.h"
 #include "../algorithms/pagerank.h"
 
-// CALL algo.pageRank('Page', 'LINKS', {iterations: 20, dampingFactor: 0.85}) YIELD node, score
+// CALL algo.pageRank(NULL, NULL)      YIELD node, score
+// CALL algo.pageRank('Page', NULL)    YIELD node, score
+// CALL algo.pageRank(NULL, 'LINKS')   YIELD node, score
 // CALL algo.pageRank('Page', 'LINKS') YIELD node, score
 
 typedef struct {
 	int n;                          // Number of nodes to rank.
 	int i;                          // Current node to return.
 	Graph *g;                       // Graph.
 	Node node;                      // Node.
-	GrB_Index *mappings;            // Mappings between extracted matrix rows and node ids.
-	LAGraph_PageRank *rankings;     // Nodes rankings.
+	GrB_Index *mapping;             // Mapping between extracted matrix rows and node ids.
+	LAGraph_PageRank *ranking;      // Nodes ranking.
 	SIValue *output;                // Array with 4 entries ["node", node, "score", score].
 } PagerankContext;
 
 ProcedureResult Proc_PagerankInvoke(ProcedureCtx *ctx, const SIValue *args) {
+	// Expecting 2 arguments.
 	if(array_len((SIValue *)args) != 2) return PROCEDURE_ERR;
-	if(!(SI_TYPE(args[0]) & SI_TYPE(args[1]) & T_STRING)) return PROCEDURE_ERR;
-
-	const char *label = args[0].stringval;
-	const char *relation = args[1].stringval;
-
-	GrB_Index n = 0;
+	// arg0 and arg1 can be either String or NULL
+	SIType arg0_t = SI_TYPE(args[0]);
+	SIType arg1_t = SI_TYPE(args[1]);
+	if(!(arg0_t & (T_STRING | T_NULL))) return PROCEDURE_ERR;
+	if(!(arg1_t & (T_STRING | T_NULL))) return PROCEDURE_ERR;
+
+	// Read arguments.
+	const char *label = NULL;    // Node filter.
+	const char *relation = NULL; // Edge filter.
+	if(arg0_t == T_STRING) label = args[0].stringval;
+	if(arg1_t == T_STRING) relation = args[1].stringval;
+
+	// Pagerank config arguments
+	int iters;         // iterations performed
+	const double tol = 1e-4; // tolerance
+	const int itermax = 100; // max iterations
+
+	GrB_Info info;
```

Comment:
Compiler warning in non-debug builds:
`warning: variable ‘info’ set but not used`
We should probably add an `UNUSED` macro to avoid these.

Reply:
👍